### PR TITLE
exit: Check if process was signaled

### PR DESCRIPTION
**Pull request by jcvenegas**
If the process was signaled with SIGTERM the exit
code reported was 255, but the correct should be 137.

This patch uses the issue based in the suggestion
in a golang issue[1]. If the process exited because was
signaled we return 128+signal.

Fixes: #144

[1] https://github.com/golang/go/issues/7938

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by sboeuf** [comment]
@jcvenegas This change looks good to me, but are you sure `processState` is not `nil` in case of `SIGKILL` ? I mean have you tested this change and does that work for you with containerd 1.0 ?

**Comment by jodh-intel** [comment]
lgtm.

Nit: shouldn't the commit mention `143` (`128 + 15` [`SIGTERM`]) rather than `137` (which would be for `SIGKILL`)?

[Approved with PullApprove](https://pullapprove.com/clearcontainers/agent/pull-request/145/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent)

**Review by grahamwhaley — COMMENTED** [review]
Nice pickup

**Comment by jodh-intel** [comment]
Random thought: now that we're using structured logging, in principle we could modify `kill_test.go` to not only check the return code, but also to ensure that -- when debug is enabled -- the correct signal/exit code is logged.

File: agent.go [hunk 1]
```diff
@@ -870,9 +870,19 @@ func (p *pod) runContainerProcess(cid, pid string, terminal bool, started chan e
 		fieldLogger.Info("Got process state")
 
 		if waitStatus, ok := processState.Sys().(syscall.WaitStatus); ok {
-			exitCode = uint8(waitStatus.ExitStatus())
-			fieldLogger.WithField("exit-code", exitCode).Info("got wait exit code")
+			exitStatus := waitStatus.ExitStatus()
+
+			if waitStatus.Signaled() {
+				exitCode = uint8(128 + waitStatus.Signal())
```
Comment (devimc) [inline agent.go]: could you please define a variable and document why 128 ?

Comment (jcvenegas) [reply agent.go]: @devimc changed to constant and documented :D

**Comment by jcvenegas** [comment]
@sboeuf  ` but are you sure processState is not nil in case of SIGKILL ? I mean have you tested this change and does that work for you with containerd 1.0 ?`

I double check and it is not nil - it returns a `255` exit code, googling I found that issue, by checking if exited by a signal seems to work well with all the cases (contained 1.0 and latest docker release).

File: agent.go [hunk 2]
```diff
@@ -67,6 +67,10 @@ const (
 	cannotGetTimeMsg   = "Failed to get time for event %s:%v"
 	pciBusRescanFile   = "/sys/bus/pci/rescan"
 	pciBusMode         = 0220
+	// If a process finish because fatal error with  signal "n"
```
Comment (sboeuf) [inline agent.go]: s/finish/terminates/
s/because fatal error with signal/because of signal/

**Comment by sboeuf** [comment]
@jcvenegas that's great if you checked ! I like the way this is handled, pretty neat!
LGTM

[Approved with PullApprove](https://pullapprove.com/clearcontainers/agent/pull-request/145/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent) [Approved with PullApprove](https://pullapprove.com/clearcontainers/agent/pull-request/145/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent)

File: agent.go (hunk 2 above)
Comment (jcvenegas) [reply agent.go]: Changed :)

**Review by sboeuf — APPROVED** [review]

**Pull request merged**